### PR TITLE
feat: WIT Interfaces for SQLite

### LINKED ISSUE #205: 🛠️ [TASK] : Documented .WIT definition mapping 1:1 to low level SQLite C API.
We need a [.wit](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) definition for Hermes that defines the core API for sqlite as documented here:

https://www.sqlite.org/cintro.html

We must use a wasm resource to wrap the database handle and the functions should otherwise be 1:1 mappings of the C API.

Specifically we need to only define The following two objects and eight methods which comprise the essential elements of the SQLite interface:

* [sqlite3](https://www.sqlite.org/c3ref/sqlite3.html) → The database connection object. Created by [sqlite3_open()](https://www.sqlite.org/c3ref/open.html) and destroyed by [sqlite3_close()](https://www.sqlite.org/c3ref/close.html).
* [sqlite3_stmt](https://www.sqlite.org/c3ref/stmt.html) → The prepared statement object. Created by [sqlite3_prepare()](https://www.sqlite.org/c3ref/prepare.html) and destroyed by [sqlite3_finalize()](https://www.sqlite.org/c3ref/finalize.html).
* [sqlite3_open()](https://www.sqlite.org/c3ref/open.html) → Open a connection to a new or existing SQLite database. The constructor for [sqlite3](https://www.sqlite.org/c3ref/sqlite3.html).
* [sqlite3_prepare()](https://www.sqlite.org/c3ref/prepare.html) → Compile SQL text into byte-code that will do the work of querying or updating the database. The constructor for [sqlite3_stmt](https://www.sqlite.org/c3ref/stmt.html).
* [sqlite3_bind()](https://www.sqlite.org/c3ref/bind_blob.html) → Store application data into [parameters](https://www.sqlite.org/lang_expr.html#varparam) of the original SQL.
* [sqlite3_step()](https://www.sqlite.org/c3ref/step.html) → Advance an [sqlite3_stmt](https://www.sqlite.org/c3ref/stmt.html) to the next result row or to completion.
* [sqlite3_column()](https://www.sqlite.org/c3ref/column_blob.html) → Column values in the current result row for an [sqlite3_stmt](https://www.sqlite.org/c3ref/stmt.html).
* [sqlite3_finalize()](https://www.sqlite.org/c3ref/finalize.html) → Destructor for [sqlite3_stmt](https://www.sqlite.org/c3ref/stmt.html).
* [sqlite3_close()](https://www.sqlite.org/c3ref/close.html) → Destructor for [sqlite3](https://www.sqlite.org/c3ref/sqlite3.html).
* [sqlite3_exec()](https://www.sqlite.org/c3ref/exec.html) → A wrapper function that does [sqlite3_prepare()](https://www.sqlite.org/c3ref/prepare.html), [sqlite3_step()](https://www.sqlite.org/c3ref/step.html), [sqlite3_column()](https://www.sqlite.org/c3ref/column_blob.html), and [sqlite3_finalize()](https://www.sqlite.org/c3ref/finalize.html) for a string of one or more SQL statements.

There are many other sqlite C functions, these are outside the scope of this task and are not to be defined in the .wit file.

Hermes allows exactly 1 sqlite db per application.
The Open call needs to be abstract, the application can set no options and can not name the sqlite file its opening, the open function just returns the resource for the application to use.
Close, closes the resource.
See https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#item-resource for how to define a resource

### PR DESCRIPTION
**Pull request by apskhem**
# Description

Create a WIT Interface to work with SQLite with fundamental functionalities from [https://www.sqlite.org/cintro.html](https://www.sqlite.org/cintro.html). Additional functions include [sqlite3_db_status](https://www.sqlite.org/c3ref/db_status.html), also adding dummy implementation on the Rust side for the interface.

## Related Issue(s)

Closes #205 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: wasm/wasi/wit/deps/hermes-sqlite/api.wit [hunk 1]
```diff
@@ -0,0 +1,274 @@
+// cspell: words ioerr cantopen toobig nolfs notadb maindbname lookaside fkey qpsg scanstatus
```
Comment (minikin) [inline wasm/wasi/wit/deps/hermes-sqlite/api.wit:1]: Can we move cpell words to project.dic, please?

**Review by minikin — CHANGES_REQUESTED** [review]
How can we validate that the wit definition is correct and matches SQLite public API?

File: wasm/wasi/wit/deps/hermes-sqlite/api.wit (hunk 1 above)
Comment (apskhem) [reply wasm/wasi/wit/deps/hermes-sqlite/api.wit]: Basically you can validate those structs and functions interfaces from this docs: [https://www.sqlite.org/cintro.html](https://www.sqlite.org/cintro.html) with additional functions that are useful [sqlite3_db_config](https://www.sqlite.org/c3ref/db_config.html), and [sqlite3_db_status](https://www.sqlite.org/c3ref/db_status.html). Error status is [here](https://www.sqlite.org/rescode.html). The `open()` function doesn't require the filepath parameter as Hermes itself internally provides for this. other functions implement interfaces according to the C interfaces provided in the docs.

**Comment by stevenj** [comment]
Add a dummy Runtime Extension implementation that doesn't do anything, but satisfies the runtime extension interface inside hermes.

**Review by stevenj — CHANGES_REQUESTED** [review]
Looks good, but to fix the CI error we need to add a dummy Runtime Extension implementation into hermes to satisfy its compile requirements.

The implementation needs no functionality, just enough definitions to allow hermes to build and link without errors.

**Review by stevenj — APPROVED** [review]
Nice.
LGTM

**Comment by stevenj** [comment]
> How can we validate that the wit definition is correct and matches SQLite public API?

It's not possible to do it automatically.
It needs a manual review.
If its not correct, it will become obvious during implementation of the runtime extension.
On my review, it looks accurate.

**Pull request merged**